### PR TITLE
PYI-587 - Create SAM template

### PR DESCRIPTION
**Pull request by TomTucka**
### What changed

Created `deploy/template.yaml` containing the sam definition to deploy this repository
Added `deploy/samconfig.toml` containing the deployment config for this repository

### Why did it change

We're moving deployments of serverless technologies to SAM

### Issue tracking

- [PYI-587](https://govukverify.atlassian.net/browse/PYI-587)
- [SAM ADR]()

### CODE REVIEW

File: deploy/samconfig.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version = 0.1
+
+
+[dev.deploy.parameters]
+stack_name = "ipv-cri-uk-passport-back-development"
+s3_bucket = "aws-sam-cli-managed-default-samclisourcebucket-ec647gpjuo2w"
+s3_prefix = "ipv-cri-uk-passport-back-development"
+region = "eu-west-2"
+capabilities = "CAPABILITY_IAM"
+parameter_overrides = "Environment=\"dev\""
```
Comment (dannyhayes-gds) [inline deploy/samconfig.toml:10]: Just needs an EOF line here

File: deploy/template.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    Type: AWS::Serverless::Api
+    Properties:
+      Name: IPV CRI UK Passport API Gateway
+      StageName: !Sub ${Environment}
+
+  IPVDcsCredentialFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "ipv-dcs-credential-${Environment}"
+      Handler: uk.gov.di.ipv.cri.passport.lambda.DcsCredentialHandler::handleRequest
```
Comment (dannyhayes-gds) [inline deploy/template.yaml:26]: I think this handler pathname is incorrect.

Should it be uk.gov.di.ipv.cri.passport.dcscredential.DcsCredentialHandler::handleRequest

File: deploy/template.yaml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+          Properties:
+            RestApiId: !Ref IPVCriUKPassportAPI
+            Path: /passport
+            Method: POST
+
+  IPVDcsStubFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "ipv-stub-dcs-${Environment}"
+      Handler: uk.gov.di.ipv.cri.passport.dcsstub.StubDcsHandler::handleRequest
```
Comment (dannyhayes-gds) [inline deploy/template.yaml:116]: Think this one is wrong as well.

uk.gov.di.ipv.cri.passport.stubdcs.StubDcsHandler::handleRequest

File: deploy/template.yaml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+          Properties:
+            RestApiId: !Ref IPVCriUKPassportAPI
+            Path: /token
+            Method: POST
+
+  IPVPassportFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "ipv-passport-${Environment}"
+      Handler: uk.gov.di.ipv.cri.passport.passport.PassportHandler::handleRequest
```
Comment (dannyhayes-gds) [inline deploy/template.yaml:84]: This one is correct but the passport.passport seems a bit weird, might mean we change that in the future maybe? But not necessarily something to worry about in this PR

Comment (TomTucka) [reply deploy/template.yaml:84]: Yeah I also hate this 😂  but it follows the pattern, I'd like us to maybe rethink the lambda name?

Comment (dannyhayes-gds) [reply deploy/template.yaml:84]: Yeah good shout, I think the lambda names in general can be better so something we can fix after

**Review by dannyhayes-gds — APPROVED** [review]

**Review by dannyhayes-gds — APPROVED** [review]

**Pull request merged**